src/ecs/Engine.hpp

```diff
@@ -0,0 +1,137 @@
+         * @param wSizeHeight height size of the window in pixel, 600 as default
+         * @param wTitle window's title, "r-type" as default
+         */
+        explicit Engine(int wSizeWidth = 800, int wSizeHeight = 600, std::string wTitle = "r-type")
+            : _worldSwitchReady(false)
+        {
+            _window = std::make_unique<sf::RenderWindow>(sf::VideoMode(wSizeWidth, wSizeHeight), wTitle);
+            _window.get()->setFramerateLimit(60);
+            ecs::World initWorld(_window);
+            ecs::Entity player = initWorld.registry.spawn_entity();
+            initWorld.registry.addComponent<ecs::component::Position>(player, {10, 10});
+            initWorld.registry.addComponent<ecs::component::Velocity>(player, {5, 5});
+            initWorld.registry.addComponent<ecs::component::Size>(player, {2, 2});
+            initWorld.registry.addComponent<ecs::component::Direction>(player, {0, 0});
+            initWorld.registry.addComponent<ecs::component::Drawable>(
+                player, {"src/demo/assets/textures/players.gif", {1, 1, 32, 16}});
+            initWorld.registry.addComponent<ecs::component::Controllable>(
+                player, {sf::Keyboard::Z, sf::Keyboard::Q, sf::Keyboard::S, sf::Keyboard::D});
+
+            ecs::Entity enemy = initWorld.registry.spawn_entity();
+            initWorld.registry.addComponent<ecs::component::Position>(enemy, {500, 500});
+            initWorld.registry.addComponent<ecs::component::Size>(enemy, {5, 5});
+            initWorld.registry.addComponent<ecs::component::Drawable>(
+                enemy, {"src/demo/assets/textures/players.gif", {1, 18, 32, 16}});
+            initWorld.registry.addComponent<ecs::component::EnemyAI>(enemy, {});
+
+            initWorld.addSystem(ecs::systems::handleSFMLEvents);
+            initWorld.addSystem(ecs::systems::handleSFMLMovements);
+            initWorld.addSystem(ecs::systems::manageClientEvents);
+            // initWorld.addSystem(ecs::systems::positionLogger);
+            initWorld.addSystem(ecs::systems::draw);
+            initWorld.addSystem(ecs::systems::movement);
+
+            _currentWorld = std::make_unique<ecs::World>(initWorld);
+        }
+
+        /**
+         * Used to get the currentWorld currently used by the ecs
+         * @return The currently used world by the ecs
+         */
+        ecs::World &getWorld() { return *_currentWorld; }
+
+        /**
+         * Set the world given as parameter to release mode
+         * @param world The world you want to set to release mode
+         */
+        void setWaitingWorld(ecs::World &world)
+        {
+            if (_waitingWorld)
+                _waitingWorld.release();
+            _waitingWorld = std::make_unique<ecs::World>(world);
+        }
+
+        /**
+         * Set the world given as parameter ready to switch for another one
+         */
+        void setWorldSwitchReady() { _worldSwitchReady = true; }
```

Comment:
consider using an `inline` func
```suggestion
        inline void setWorldSwitchReady() { _worldSwitchReady = true; }
```